Feature/sref metplus5

BinbinZhou-NOAA opened the pull request:
## Pull Request Testing ##

- [y ] Describe testing already performed for this Pull Request:</br>

- [ y] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

- [ y] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ y] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [y ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [y ] Review the source issue metadata (required labels, projects, and milestone).
- [ y] Complete the PR description above.
- [ y] Ensure the PR title matches the feature branch name.
- [ y] Check the following:
- [y ]  Instructions provided on how to run
- [ y]  Developer's name is replaced by ${user} where necessary throughout the code
- [y ]  Check that the ecf file has all the proper definitions of variables
- [y ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ y]  Check to see that the output directory structure is followed
- [ y]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ y] After submitting the PR, select **Development** issue with the original issue number.
- [ y] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [y ] Close the linked issue.

PerryShafran-NOAA committed:
Update README.md

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
First update

BinbinZhou-NOAA committed:
update run.ver

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
delete __pycache__

BinbinZhou-NOAA committed:
new update

BinbinZhou-NOAA commented:
**Part 1/2 Instruction for stat part testing**

Testing ecf directory:
  /lfs/h2/emc/vpppg/noscrub/$USER/EVS/ecf/mesoscale/stats

  There are 3 ecf files:
   jevs_sref_cnv_stats.ecf
   jevs_sref_grid2obs_stats.ecf
   jevs_sref_precip_stats.ecf

   Tese 3 ecf jobs have no relationship, but jevs_sref_cnv_stats.ecf must be run first about 30min before
   jevs_sref_grid2obs_stats.ecf and jevs_sref_precip_stats.ecf.
   jevs_sref_grid2obs_stats.ecf and jevs_sref_precip_stats.ecf can be run at same time


Note: jevs_sref_cnv_stats.ecf still set  USE_CASE=grid2obs
      In  jevs_sref_cnv_stats.ecf, set just_cnv=yes while jevs_sref_grid2obs_stats.ecf set just_cnv=no

Before test, **please first compile the Fortran code** in /sorc/evs_sref_adjust_precip24_time.fd
  by running the script compile_evs_global_ens_sorc.sh in /sorc
  This code is for the interpolation of 09Z and 15Z  SREF's 24hr APCP mean fields and get 12Z
  SREF's 24hr APCP. The results will be used to generate SREF's 24hr APCP spatial map.


Submit jobs:
  qsub jevs_sref_cnv_stats.ecf
  **sleep 1800**
  qsub jevs_sref_grid2obs_stats.ecf
  qsub jevs_sref_precip_stats.ecf

  Note: jevs_sref_grid2obs_stats.ecf must wait for finishing of evs_sref_cnv_stats.ecf, so wait 1800 sec

  Note: since "#PBS -j oe" is set, the stdout file will be in the submit directory
   /lfs/h2/emc/vpppg/noscrub/$USER/EVS/ecf/mesoscale/stats

The running directory (i.e. $DATA) will be in /lfs/h2/emc/ptmp/${USER}/evs/tmpnwprd:

  evs_sref_grid2obs_stats.$jobid for both jevs_sref_cnv_stats.ecf and jevs_sref_grid2obs_stats.ecf
  evs_sref_precip_stats.$jobid

  jobid is automatically created ($$) during the run

The running has 4 stages for each job

 (0) Check validation data, including observation data and SREF 26 ensemble member files
      If observation data is missing, the run will stop
      If any sref member(s) is(are) missing, the run will stop
      If sref member files are missing for all forecast hours, the run will stop

    IF none of above happen, go following stages

 (1) Run METplus to process PB2NC prepbufr files (grid2obs) or PCP_Combine 6hr CCPA  data and sref's 6h APCP data files (precip)

     After finished,
     the prepbufr files are in  $DATE/prepbufr.vdate sub-directory
     the pcp_combine files are in $DATA/ccpa.vdate, sref.initdate1, sref.initdate2, sref.initdate3, sref.initdate4, and sref.initdate5
     Since there will be 5 days and 26 sref member's 6hr APCP need to be generated, it will take a while to complete this procedure

 (2) Run METplus to generate stat files

     For grid2obs
       All of output stat files will be in the $DATA/grid2obs
     For precip
       All of output stat files will be in  the $DATA/precip

     After the stage (1) is finished,

        for grid2obe following 2 sub-job scripts are generated for MPI run
           run_sref_g2o_CONUS.prepbufr.fhr1.sh
           run_sref_g2o_CONUS.prepbufr.fhr2.sh

        for precip following 2 sub-job scripts are generated for MPI run
           run_sref_mpi_CONUS.ccpa.fhr1.sh
           run_sref_mpi_CONUS.ccpa.fhr2.sh

    The METplus output_base for each sub-jobs are in grid2obs or precip:
      for grid2obs:
          run_sref_g2o_CONUS.prepbufr.fhr1
          run_sref_g2o_CONUS.prepbufr.fhr2

      for precip
          run_sref_mpi_CONUS.ccpa.fhr1
          run_sref_mpi_CONUS.ccpa.fhr2

      Each output_base is correspondent to a parallel job.

    In each job's output_base directory, there are 3 sub directories
          logs, stats, and tmp

    The small stat files are stored in the stats sub-directory.  For grid2obs and precip , the small stat files have 3 types:
          (1) ensemble_stat*.stat
          (2) point_stat*CNV*.stat
          (3) point_stat*MEAN*.stat (grid2obs) or grid_stat*MEAN*.stat (precip)
          (4) point_stat*PROB*.stat (grid2obs) or grid_stat*PROB*.stat (precip)

    The small stat files are stored in directory /lfs/h2/emc/vpppg/noscrub/binbin.zhou/com/evs/v1.0/stats/mesoscale/atmos.$VDATE/sref
       grid2obs and precip, respectively


 (3) Combine the small stat files to a large stat file

   The small stat files  are then combined (by StatAnalysis tool) into 2  large stat files
           sref_grid2obs_v$DATE.stat
           sref_precip_v$VDATE.stat

   This large stat file is stored in the directory
      /lfs/h2/emc/vpppg/noscrub/binbin.zhou/com/evs/v1.0/stats/mesoscale/sref.$VDATE

  The StatAnalysis METplus output_base directory is $DATA/gather, in which all of running info are stored

 **Part 2/2 Instruction for plot part testing:**

Testing ecf directory:
  /lfs/h2/emc/vpppg/noscrub/$USER/EVS/ecf/mesoscale/plots

  There are 4 ecf files:
   jevs_sref_grid2obs_past90days_plots.ecf
   jevs_sref_precip_spatial_plots.ecf
   jevs_sref_cnv_past90days_plots.ecf
   jevs_sref_precip_past90days_plots.ecf

   These 4 ecf jobs are independent and can be submit at same time

Submit jobs
   qsub jevs_sref_grid2obs_past90days_plots.ecf
   qsub jevs_sref_precip_spatial_plots.ecf
   qsub jevs_sref_cnv_past90days_plots.ecf
   qsub jevs_sref_precip_past90days_plots.ecf

  Note: since "#PBS -j oe" is set, the stdout file will be in the submit directory
   /lfs/h2/emc/vpppg/noscrub/$USER/EVS/ecf/mesoscale/plots

The running directory (i.e. $DATA) will be in /lfs/h2/emc/${USER}/evs/tmpnwprd

There is a script called submit_all_plots.sh. By running this script, all 4 jobs
can be submitted.

The generated plots are tarred in one tar files for each job and is sent to the COMOUT directory
/lfs/h2/emc/vpppg/noscrub/${USER}/com/evs/v1.0/plots/mesoscale/atmos.$VDATE:
  evs.plots.sref.cnv.v.90.tar
  evs.plots.sref.grid2obs.v${VDAT}.90.tar
  evs.plots.sref.precip.v.${VDAT}.90.both.valid.times.tar
  evs.plots.href.precip.spatial.map.v${VDAT}.tar

PerryShafran-NOAA commented on ecf/mesoscale/stats/jevs_sref_cnv_stats.ecf:
Hi, Binbin.  I see you changed back the PBS -N attribute from jevs to evs.  Please, let's keep this as jevs - this needs to be the exact name as the job.

Thanks!

Perry

BinbinZhou-NOAA replied:
OK, I have changed back to jevs...

PerryShafran-NOAA commented on ecf/mesoscale/stats/submit_sref_jobs.sh:
Remove file from directory.

BinbinZhou-NOAA replied:
Sure, I have deleted as well

PerryShafran-NOAA commented on jobs/mesoscale/stats/JEVS_MESOSCALE_STATS:
I'm looking at this, and noticing that the sections regarding the NAM and RAP have been removed.  Please do not do this.  Please be sure you are starting with the jobs file in the authoritative repo.

BinbinZhou-NOAA replied:
Sure, I have updated it to use the autho's one

malloryprow replied:
I think the removal of lines 40 and 41 might break the NAM and RAP stats. I've copied Roshan's style in the ecf scripts and it has DATAROOTtmp set.

malloryprow replied:
But perhaps DATAROOTtmp shouldn't be used at all. If so Roshan and I will need to update our ecf scripts.

PerryShafran-NOAA commented on parm/metplus_config/machine.conf:
Please be sure you are using the authoritative version of machine.conf in your filesystem.  Thanks!

BinbinZhou-NOAA replied:
I have updated to use authos's one

BinbinZhou-NOAA committed:
Update ecf and scripts

BinbinZhou-NOAA committed:
Update ecf and JOB card

BinbinZhou-NOAA commented:
Perry,

  All of above issues have been fixed and updated in the github
feature/sref_metplus5
(1)  PBS -N attribute from evs to PBS -N jevs
(2) use autho's J JOB file for stats
(3) use autho's machine.cong
(4) plot tar file name from plots.... to evs.plots ..

Please update them in your testing

Binbin




On Mon, Apr 10, 2023 at 11:08 AM PerryShafran-NOAA ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On ecf/mesoscale/stats/jevs_sref_cnv_stats.ecf
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1161798539>:
>
> Hi, Binbin. I see you changed back the PBS -N attribute from jevs to evs.
> Please, let's keep this as jevs - this needs to be the exact name as the
> job.
>
> Thanks!
>
> Perry
> ------------------------------
>
> On ecf/mesoscale/stats/submit_sref_jobs.sh
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1161798943>:
>
> Remove file from directory.
> ------------------------------
>
> On jobs/mesoscale/stats/JEVS_MESOSCALE_STATS
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1161800371>:
>
> I'm looking at this, and noticing that the sections regarding the NAM and
> RAP have been removed. Please do not do this. Please be sure you are
> starting with the jobs file in the authoritative repo.
> ------------------------------
>
> On parm/metplus_config/machine.conf
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1161800941>:
>
> Please be sure you are using the authoritative version of machine.conf in
> your filesystem. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#pullrequestreview-1377664992>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFK3DZTJIN2DRS5XVD3XAQO4XANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
chnage COMIN, COMOUT

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
update J-job and ecf

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
add missing data alert

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
Change run.ver to run.ver.metplus5.0.0

BinbinZhou-NOAA commented:
Mallory,

  In my new version of
https://github.com/BinbinZhou-NOAA/EVS/blob/feature/sref_metplus5/jobs/mesoscale/stats/JEVS_MESOSCALE_STATS
DATA_IN has been removed to make consistent to other mesoscale models, and
DATA is defined in the ecf files, e.g.
https://github.com/BinbinZhou-NOAA/EVS/blob/feature/sref_metplus5/ecf/mesoscale/stats/jevs_sref_grid2obs_stats.ecf


Binbin

On Thu, Apr 27, 2023 at 4:20 PM Mallory Row ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On jobs/mesoscale/stats/JEVS_MESOSCALE_STATS
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1179638313>:
>
> I think the removal of lines 40 and 41 might break the NAM and RAP stats.
> I've copied Roshan's style in the ecf scripts and it has DATAROOTtmp set.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1179638313>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFI5GJYIU2YYFPDFXNLXDLIHHANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
Update to remove sh in execution of scripts

BinbinZhou-NOAA committed:
add jobid in ecf files

AliciaBentley-NOAA commented:
@malloryprow Did you questions get answered by @BinbinZhou-NOAA's latest response or are changes potentially still needed to your/Roshan's EVS code?

malloryprow commented:
@AliciaBentley-NOAA Yup, all good!

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
change Pb2nc_obsGFS_Prepbufr.cong Pb2nc_obsGFS_Prepbufr.conf

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA committed:
update evs_href_preppare.sh for href PR for cong file change

ShelleyMelchior-NOAA commented:
I have cloned BinbinZhou-NOAA/EVS and am working from `feature/sref_metplus5`

ShelleyMelchior-NOAA commented:
I compiled `sorc/evs_sref_adjust_precip24_time.fd`. One little hang up is I had to manually create the `exec/` dir.

I ran `ecf/mesoscale/stats/jevs_sref_cnv_stats.ecf`.
logifle: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_cnv_stats.o57228377`
data: `COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale/`

I assume the `sleep 1800` is to allow enough time for `jevs_sref_cnv_stats.ecf` to complete.

I ran `ecf/mesoscale/stats/jevs_sref_grid2obs_stats.ecf`.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_g2o_stats.o57230802`
data: `COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale`

I ran `ecf/mesoscale/stats/jevs_sref_precip_stats.ecf`.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_precip_stats.o57235108`
data: `COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/mesoscale`

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA  -- please update your `feature/sref_metplus5` branch to sync with the updated auth EVS develop. I just merged Shannon's latest PR.

BinbinZhou-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/sref_metplus5

BinbinZhou-NOAA commented:
Shelley,

  Thanks!
  I have checked the  stat output files from your sref PR parallel runs,
showing they are all fine,
exactly the same as the same files from my runs.

  I also have  synced to merge the auth repo into the feature/sref_metplus5

  One reminder:  if you try to run the plot generation ecf, you should
have  enough historical stat files of  BOTH
sref and gefs. If you have no such historical stat files,  you can use my
files from both
/lfs/h2/emc/vpppg/noscrub/binbin.zhou/evs/v1.0/stats/mesoscale
and
/lfs/h2/emc/vpppg/noscrub/binbin.zhou/evs/v1.0/stats/global_ens

Binbin




On Fri, May 5, 2023 at 7:30 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- please update
> your feature/sref_metplus5 branch to sync with the updated auth EVS
> develop. I just merged Shannon's latest PR.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1536902515>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFPX7F6SQXRMNTWOLKTXEWERNANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented on sorc/evs_sref_adjust_precip24_time.fd/makefile:
```diff
@@ -0,0 +1,21 @@
+
+FC=   ftn -convert big_endian    
+INC= ${G2_INC4}
+BINDIR= ../../exec
```
I suggest adding a line to force the creation of $BINDIR in case the directory does not exist yet. If it does not exist, the executable does not get moved and installed in line 19. I suggest the following be added just after line 10.
`mkdir -p $BINDIR`

ShelleyMelchior-NOAA commented:
I modified the 4 ecf plots scripts to point to my $HOMEevs as well as to point $COMIN into Binbin's gathered stats directory. The 4 resultant tar files reside here:
`COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230505`
I only untarred `evs.plots.sref.cnv.90.v20230505.tar`. I transferred the png impages to https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/sref/ for easier viewing.
@BinbinZhou-NOAA , please take a closer look at the rest of the tar files.

Output log files are located in `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out`:
`jevs_sref_grid2obs_plot.o57350979`
`jevs_sref_precip_spatial_plot.o57351430`
`jevs_sref_cnv_plot.o57351500`
`jevs_sref_precip_plot.o57351524`

BinbinZhou-NOAA commented:
Shelley,

  Greadt! I have reviewed all of the 4 plot tar files in your output
directory, showing all are in good shape, and
exactly the same as my files.   A part of the plots  in your website  also
look good.
To  show all of plots you just generated,  I  created another website
       https://www.emc.ncep.noaa.gov/bzhou/evs_plots/sref_shelley
There are 4 sub-directoies inside, each showing different types of plots.
Please double check them to see if all are fine.

Thanks!

Binbin


On Sat, May 6, 2023 at 8:34 PM Shelley Melchior ***@***.***>
wrote:

> I modified the 4 ecf plots scripts to point to my $HOMEevs as well as to
> point $COMIN into Binbin's gathered stats directory. The 4 resultant tar
> files reside here:
>
> COMOUT=/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230505
> I only untarred evs.plots.sref.cnv.90.v20230505.tar. I transferred the
> png impages to https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/sref/
> for easier viewing.
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , please take a
> closer look at the rest of the tar files.
>
> Output log files are located in /lfs/h2/emc/ptmp/shelley.melchior/EVS_out:
> jevs_sref_grid2obs_plot.o57350979
> jevs_sref_precip_spatial_plot.o57351430
> jevs_sref_cnv_plot.o57351500
> jevs_sref_precip_plot.o57351524
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1537253537>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFIQI5ZVYCG7IO6BA5LXE3UZFANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Great, @BinbinZhou-NOAA -- 2 more things.

1. I chose random plots to look at. Is this one correct? The legend/key doesn't seem to match what is being plotted in terms of colors and line styles? It could be fine and I am just not understanding the plot correctly. https://www.emc.ncep.noaa.gov/bzhou/evs_plots/sref_shelley/grid2obs/evs.sref.rmse_spread.ugrd_10m.last90days.fhrmean.valid_00z_06z_12z_18z.conus_glb.png
2.  Please update `sorc/evs_sref_adjust_precip24_time.fd/makefile` with my recommendation at line 10.

BinbinZhou-NOAA commented:
Shelley,

  Thanks for further double checking the plots. The black color for GEFS
should be ok, while the color red
for SREF, just following the setting of Marcel's code.  Since just 2
models, visually it should be fine.
If it is not fine, I can modify it for SREF.
  Yes, I have modified the sorc/evs_sref_adjust_precip24_time.fd/makefile,
but still update
on github

Binbin



On Mon, May 8, 2023 at 11:46 AM Shelley Melchior ***@***.***>
wrote:

> Great, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- 2 more
> things.
>
>    1. I chose random plots to look at. Is this one correct? The
>    legend/key doesn't seem to match what is being plotted in terms of colors
>    and line styles? It could be fine and I am just not understanding the plot
>    correctly.
>    https://www.emc.ncep.noaa.gov/bzhou/evs_plots/sref_shelley/grid2obs/evs.sref.rmse_spread.ugrd_10m.last90days.fhrmean.valid_00z_06z_12z_18z.conus_glb.png
>    2. Please update sorc/evs_sref_adjust_precip24_time.fd/makefile with
>    my recommendation at line 10.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1538617909>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFJ4GNATFVDNKXKWO4LXFEIN3ANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA committed:
update sorc/evs_sref_adjust_precip24_time.fd/makefile

BinbinZhou-NOAA commented:
I just updated the makefile on the github sref_metplus5 branch

Binbin

On Mon, May 8, 2023 at 12:18 PM Binbin Zhou - NOAA Affiliate <
***@***.***> wrote:

> Shelley,
>
>   Thanks for further double checking the plots. The black color for GEFS
> should be ok, while the color red
> for SREF, just following the setting of Marcel's code.  Since just 2
> models, visually it should be fine.
> If it is not fine, I can modify it for SREF.
>   Yes, I have modified the sorc/evs_sref_adjust_precip24_time.fd/makefile,
> but still update
> on github
>
> Binbin
>
>
>
> On Mon, May 8, 2023 at 11:46 AM Shelley Melchior ***@***.***>
> wrote:
>
>> Great, @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> -- 2 more
>> things.
>>
>>    1. I chose random plots to look at. Is this one correct? The
>>    legend/key doesn't seem to match what is being plotted in terms of colors
>>    and line styles? It could be fine and I am just not understanding the plot
>>    correctly.
>>    https://www.emc.ncep.noaa.gov/bzhou/evs_plots/sref_shelley/grid2obs/evs.sref.rmse_spread.ugrd_10m.last90days.fhrmean.valid_00z_06z_12z_18z.conus_glb.png
>>    2. Please update sorc/evs_sref_adjust_precip24_time.fd/makefile with
>>    my recommendation at line 10.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1538617909>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQBMPFJ4GNATFVDNKXKWO4LXFEIN3ANCNFSM6AAAAAAWVXCAZA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
> Binbin Zhou
>
> Physical Scientist
>
> Lynker at NOAA/NWS/NCEP/EMC
>
> 5830 University Research Ct.
>
> College Park, MD 20740
>
> ***@***.***
>
> 301-683-3683
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA reviewed (commented):
I am going to approve this PR with a few items of caution to be addressed for the next PR:

- Work w/ Alicia to discuss a plan to improve the legend for the plots that show RMSE
- Add a trap to the stats ecf scripts such that the process cannot continue if $maillist is not defined

ShelleyMelchior-NOAA commented on ush/mesoscale/evs_prepare_sref.sh:
Need `#!/bin/ksh` or `#!/bin/bash` on top line

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/df_preprocessing.py:
```diff
@@ -0,0 +1,306 @@
+###############################################################################
```
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/lead_average.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/performance_diagram.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/plot_util.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/plotter.py:
```diff
@@ -0,0 +1,163 @@
+import matplotlib.pyplot as plt
```
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/prune_stat_files.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/roc_curve.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/settings.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_cnv_py/stat_by_level.py:
For python scripts you must use the following in the first line:
`#! /usr/bin/env python3`
NCO standards requires the interpreter be stated in line 1.

ShelleyMelchior-NOAA reviewed (commented):
Please review all python scripts in `/ush/mesoscale/ush_sref_plot*/*.py` to make sure the first line indicates the python interpreter:
`#! /usr/bin/env python3`

(I started to point this out in each individual python script in `ush/mesoscale/ush_sref_plot*/*.py` but abandoned b/c there are too many.)

ShelleyMelchior-NOAA commented on versions/run.ver.old:
Please remove this file from `feature/sref_metplus5`. This file, `versions/run.ver.old` is not in the auth repo develop and I do not want to introduce it.

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA , I know I just added a flurry of activity to this PR ... so, in SUMMARY ...

- update the ush python scripts to add interpreter to top of script
- add trap to ecf stats scripts if $maillist is not defined

The key/legend for plots w/ RMSE can be updated in a later PR.

(Generally how I work a PR is run the code first and once everything looks good based on developer feedback I then look more closely at the code itself.)

BinbinZhou-NOAA commented:
Shelley,

  Sure, the file run.ver.old will be removed.
For py scripts, they are originally copied from Marcel's codes. Marcel and
other people may  still not notice this
rule.  So please bring this notice to the next regional and global
bi-weekly meetings.

Binbin

On Mon, May 8, 2023 at 2:09 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I know I just
> added a flurry of activity to this PR ... so, in SUMMARY ...
>
>    - update the ush python scripts to add interpreter to top of script
>    - add trap to ecf stats scripts if $maillist is not defined
>
> The key/legend for plots w/ RMSE can be updated in a later PR.
>
> (Generally how I work a PR is run the code first and once everything looks
> good based on developer feedback I then look more closely at the code
> itself.)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1538817198>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFK56EIRSITOQMT6V3LXFEZEVANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

BinbinZhou-NOAA commented:
Shelley,

For  "add trap to ecf stats scripts if $maillist is not defined"

 Can you give me an example?

Thanks!

Binbin


On Mon, May 8, 2023 at 2:09 PM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , I know I just
> added a flurry of activity to this PR ... so, in SUMMARY ...
>
>    - update the ush python scripts to add interpreter to top of script
>    - add trap to ecf stats scripts if $maillist is not defined
>
> The key/legend for plots w/ RMSE can be updated in a later PR.
>
> (Generally how I work a PR is run the code first and once everything looks
> good based on developer feedback I then look more closely at the code
> itself.)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1538817198>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFK56EIRSITOQMT6V3LXFEZEVANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Absolutely! See any of Logan's ecf prep scripts in `ecf/cam/prep`. For example see lines 82-91 of this code: https://github.com/NOAA-EMC/EVS/blob/1bf935ddfc359496b3354c13a8905f7b7751285d/ecf/cam/prep/jevs_namnest_severe_prep.ecf#L82

ShelleyMelchior-NOAA commented:
Regarding the ush python code, OK. Marcel is out of the office until Friday, May 12. We can let that item go unaddressed for this PR and engage with him upon his return. It will need to be addressed before code hand off to NCO.

BinbinZhou-NOAA committed:
update py by adding head and add mailing list trap

BinbinZhou-NOAA committed:
update ush/mesoscale/ush_sref_plot_cnv_py/threshold_average.py

BinbinZhou-NOAA commented:
Shelley,

  I have

(1) modified all py scripts by adding the head,
(2) added the head in  evs_prepare_sref.sh file , and
(3) added the mailing list trap the stats ecf files

And  updated these on the github sref_metplus5

Binbin

On Mon, May 8, 2023 at 3:17 PM Shelley Melchior ***@***.***>
wrote:

> Regarding the ush python code, OK. Marcel is out of the office until
> Friday, May 12. We can let that item go unaddressed for this PR and engage
> with him upon his return. It will need to be addressed before code hand off
> to NCO.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1538911462>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFN26CEAOICQBPVBZELXFFBD5ANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
Thank you Binbin! 
I am pretty sure Logan and Alicia plan to inform about the interpreter statements at their meetings this week. You are finding out about it earlier than the rest b/c of the nature of the timing of your PRs.

I will re-run one of the stats+plots processes to make sure nothing is amiss after the noted changes. I think I'll pick precip.

AliciaBentley-NOAA commented:
Hi everyone,

Please see Logan's email titled "Reminder about a few NCO Implementation
Standards" from Thu, Apr 27, 2:38 PM for more information on interpreter
statements! Thanks!

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬


On Mon, May 8, 2023 at 4:31 PM Shelley Melchior ***@***.***>
wrote:

> Thank you Binbin!
> I am pretty sure Logan and Alicia plan to inform about the interpreter
> statements at their meetings this week. You are finding out about it
> earlier than the rest b/c of the nature of the timing of your PRs.
>
> I will re-run one of the stats+plots processes to make sure nothing is
> amiss after the noted changes. I think I'll pick precip.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1539002755>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APUWOBBGB6T2RY5ZNQRPDTLXFFJ2FANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ShelleyMelchior-NOAA commented:
I updated my clone of `BinbinZhou-NOAA:feature/sref_metplus5`
I wasn't sure if I needed to re-run `jevs_sref_cnv_stats.ecf` in order to support a run of `jevs_sref_precip_stats.ecf`, so I ran both to be safe.
output logfiles: 
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_cnv_stats.o57561734`
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_precip_stats.o575645` [Note: still running as of time of this PR comment]

I also ran `jevs_sref_prepcip_spatial_plots.ecf`.
output logfile:
`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_precip_spatial_plot.o57565351`

For all runs, stats and plots are available in COMOUT: `/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/`

BinbinZhou-NOAA commented:
Shelley,

  It seems your runs are still not finished. After the runs  finish, I will
double check them

Thanks!

Binbin

On Tue, May 9, 2023 at 10:21 AM Shelley Melchior ***@***.***>
wrote:

> I updated my clone of BinbinZhou-NOAA:feature/sref_metplus5
> I wasn't sure if I needed to re-run jevs_sref_cnv_stats.ecf in order to
> support a run of jevs_sref_precip_stats.ecf, so I ran both to be safe.
> output logfiles:
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_cnv_stats.o57561734
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_precip_stats.o575645
> [Note: still running as of time of this PR comment]
>
> I also ran jevs_sref_prepcip_spatial_plots.ecf.
> output logfile:
>
> /lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_sref_precip_spatial_plot.o57565351
>
> For all runs, stats and plots are available in COMOUT:
> /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1540225162>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFN2MXWG3N6P73FX6A3XFJHFTANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented:
@BinbinZhou-NOAA , the SREF precip stats job is complete now.

BinbinZhou-NOAA commented:
Shelley,

   I just double checked your stat file and plot tar files. Both are good,
exactly the same as my files.

Thanks!

Binbin

On Tue, May 9, 2023 at 11:05 AM Shelley Melchior ***@***.***>
wrote:

> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> , the SREF precip
> stats job is complete now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#issuecomment-1540328327>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFIJXEK3OSVXEXDS6QLXFJMMHANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA commented on versions/run.ver:
```diff
@@ -20,7 +20,7 @@ export jasper_ver=2.0.25
 export libjpeg_ver=9c
 export libpng_ver=1.6.37
-export met_ver=11.0.2
+export met_ver=11.0.1
```
@BinbinZhou-NOAA please change this to 11.0.2, as it is in the auth EVS develop.

ShelleyMelchior-NOAA commented on ush/mesoscale/ush_sref_plot_py/.plot_util.py.swp:
@BinbinZhou-NOAA  Please remove this hidden binary file from your feature branch.

BinbinZhou-NOAA committed:
delete ush/mesoscale/ush_sref_plot_py/.plot_util.py.swp and modify  versions/run.ver

BinbinZhou-NOAA commented:
Sheeley,

  Done!

Binbin

On Tue, May 9, 2023 at 11:25 AM Shelley Melchior ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On ush/mesoscale/ush_sref_plot_py/.plot_util.py.swp
> <https://github.com/NOAA-EMC/EVS/pull/96#discussion_r1188778432>:
>
> @BinbinZhou-NOAA <https://github.com/BinbinZhou-NOAA> Please remove this
> hidden binary file from your feature branch.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/EVS/pull/96#pullrequestreview-1418911577>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQBMPFKVT6GSAGPJXJ562V3XFJOW5ANCNFSM6AAAAAAWVXCAZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Binbin Zhou

Physical Scientist

Lynker at NOAA/NWS/NCEP/EMC

5830 University Research Ct.

College Park, MD 20740

***@***.***

301-683-3683

ShelleyMelchior-NOAA reviewed (approved):
Everything is buttoned up and ready for approval.
